src/main/java/org/icgc/argo/program_service/grpc/ProgramServiceImpl.java

```diff
@@ -185,35 +203,29 @@ public void joinProgram(JoinProgramRequest request, StreamObserver<JoinProgramRe
 
   @Override
   public void listPrograms(Empty request, StreamObserver<ListProgramsResponse> responseObserver) {
-    val programEntities = programService.listPrograms();
+      List<ProgramEntity> programEntities = programService.listPrograms().stream().
+        filter(p -> authorizationService.canRead(p.getShortName())).
```

Comment:
based on the matrix, `listPrograms` only allows dcc admins, so should be calling `requireDCCAdmin`.

Reply:
No, because listPrograms() is supposed to list all of *YOUR* programs.
DCCAdmin gets all of them.

Reply:
The DCC Admin might not have a list of all programs in their scopes, since they'll just have the PROGRAM_SERVICE.WRITE scope... so won't their `programEntities` list be empty?

Reply:
DCCAdmin access is (currently) handled by checking for the ADMIN vs USER type in the Ego authorization token.

Reply:
yes and I believe this is not the correct way to handle it.

Reply:
but either way, they wouldn't have the full list of programs in their scopes, so I think my question still stands?

Reply:
If they have DCCAdmin access, they don't need to have any programs in their scopes -- the auth check passes.